src/main/java/com/isaacbrodsky/h3measurements/IndexRandom.java

```diff
@@ -0,0 +1,72 @@
+package com.isaacbrodsky.h3measurements;
+
+import java.io.IOException;
+import java.util.Random;
+import java.util.concurrent.ThreadLocalRandom;
+
+import com.google.common.geometry.S2CellId;
+import com.isaacbrodsky.h3measurements.h3.H3Provider;
+import com.isaacbrodsky.h3measurements.s2.S2Provider;
+import com.uber.h3core.H3Core;
+
+/**
+ * Provides DGGS indexes chosen randomly by building valid index values
+ */
+public class IndexRandom implements H3Provider, S2Provider {
+    private final H3Core h3;
+    private final Random random;
+
+    public IndexRandom() throws IOException {
+        this(H3Core.newInstance(), ThreadLocalRandom.current());
+    }
+
+    public IndexRandom(H3Core h3, Random random) {
+        this.h3 = h3;
+        this.random = random;
+    }
+
+    @Override
+    public long h3Index(int res) {
```

Comment:
I'd suggest waiting for https://github.com/uber/h3/pull/719 - once this lands, this is as simple as the S2 version (pick a random base cell, pick a random child by numeric index). One consideration there is that we would slightly over-sample pentagon base cells (they have the same likelihood of being selected, but they have fewer children) - we could potentially weight them lower to compensate.